segmentstore/storage/impl/src/main/java/io/pravega/segmentstore/storage/impl/hdfs/HDFSStorage.java

```diff
@@ -136,52 +139,187 @@ public void initialize(long epoch) {
             conf.set("dfs.client.block.write.replace-datanode-on-failure.policy", "NEVER");
         }
 
-        this.context = new FileSystemOperation.OperationContext(epoch, openFileSystem(conf), this.config);
+        this.context = new OperationContext(epoch, openFileSystem(conf), this.config);
         log.info("Initialized (HDFSHost = '{}', Epoch = {}).", this.config.getHdfsHostURL(), epoch);
     }
 
-    protected FileSystem openFileSystem(Configuration conf) throws IOException {
-        return FileSystem.get(conf);
+    @Override
+    public SegmentProperties getStreamSegmentInfo(String streamSegmentName) throws StreamSegmentException {
+        long traceId = LoggerHelpers.traceEnter(log, "getStreamSegmentInfo", streamSegmentName);
+        try {
+            return Retry.withExpBackoff( 1, 5, MAX_ATTEMPT_COUNT)
```

Comment:
Store this, all the way up to `run` (excluding it) in a final static variable in the class.

Reply:
Fixed.